logos/src/logos/pipeline/base_scheduler.py

```
@@ -246,28 +278,42 @@ def reevaluate_model_queues(self, model_name: str) -> None:
             if ptype != "logosnode":
                 continue
-            # Check if this provider now has capacity for the model
+
+            # Use lane readiness as the primary check — it reads the runtime
+            # snapshot directly, bypassing the 5s refresh_interval cache in
+            # _loaded_models that can be stale right after a cold load confirms.
+            try:
+                if not self._logosnode.is_model_lane_ready(model_id, provider_id):
+                    continue
+            except Exception:
+                continue
```

Ruff (0.15.9)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)